hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -311,8 +409,12 @@ object InferPType {
         InferPType(body, env.bind(contextsName -> contexts.pType2, globalsName -> globals.pType2))
         PArray(body.pType2)
       }
+        // TODO: should we read metadata here to get requiredeness for the row?
```

Comment:
I think the correct thing to do here (within the context of the intended use case, which is a partition operation for a lowered table) is for ReadPartition to know its own PType. (in terms of the requiredness of the stream, current behavior propagates the missingness of the path so it should be required if the path IR type is required).